frontend/views/containers/settings/UserProfile.vue

```diff
@@ -170,43 +152,47 @@ export default {
       sbp('okTurtles.events/emit', OPEN_MODAL, mode)
       return false
     },
-    async save () {
-      if (this.ephemeral.newPicture) {
-        try {
-          this.form.picture = await imageUpload(this.form.picture)
-          this.ephemeral.newPicture = false
-        } catch (error) {
-          console.error(error)
-          this.ephemeral.errorMsg = L('Failed to upload user picture')
-          return false
+    async fileChange (fileList) {
+      if (!fileList.length) return
+      const fileReceived = fileList[0]
+      let picture
+
+      this.$refs.picture.setFromBlob(fileReceived)
+
+      try {
+        picture = await imageUpload(fileReceived)
+      } catch (error) {
+        console.error(error)
+        this.$refs.pictureMsg.danger(`${L('Failed to upload avatar. Please, try again.')} ${error.message}`)
+        return false
+      }
+
+      await this.setAttributes({ picture }, 'pictureMsg', L('Picture updated!'))
```

Comment:
If this fails we probably want to reset the picture in the UI to whatever it was before so that it doesn't look like the upload succeeded.

Reply:
Another approach: change the picture in the UI only if this succeeds.

Reply:
Yes, I would say that the picture should only be updated if the upload was successful :)